src/main/java/edu/ie3/datamodel/io/source/csv/CsvTimeSeriesSource.java

```diff
@@ -122,24 +134,28 @@ public Optional<V> getValue(ZonedDateTime time) {
    * @throws SourceException If the file cannot be read properly
    * @return An option onto an individual time series
    */
-  private IndividualTimeSeries<V> buildIndividualTimeSeries(
+  protected IndividualTimeSeries<V> buildIndividualTimeSeries(
       UUID timeSeriesUuid,
       String filePath,
       Function<Map<String, String>, Optional<TimeBasedValue<V>>> fieldToValueFunction)
       throws SourceException {
     try (BufferedReader reader = dataSource.connector.initReader(filePath)) {
-      Set<TimeBasedValue<V>> timeBasedValues =
-          dataSource
-              .buildStreamWithFieldsToAttributesMap(TimeBasedValue.class, reader)
-              .map(fieldToValueFunction)
-              .flatMap(Optional::stream)
-              .collect(Collectors.toSet());
-
+      Set<TimeBasedValue<V>> timeBasedValues = getTimeBasedValueSet(fieldToValueFunction, reader);
       return new IndividualTimeSeries<>(timeSeriesUuid, timeBasedValues);
     } catch (FileNotFoundException e) {
       throw new SourceException("Unable to find a file with path '" + filePath + "'.", e);
     } catch (IOException e) {
       throw new SourceException("Error during reading of file'" + filePath + "'.", e);
     }
   }
+
+  private Set<TimeBasedValue<V>> getTimeBasedValueSet(
+      Function<Map<String, String>, Optional<TimeBasedValue<V>>> fieldToValueFunction,
```

Comment:
Isn't `fieldToValueFunction` always the function `createTimeBasedValue`?

Reply:
Ok what you did there was also helpful, it's not what I meant though :D Not that important though